homeassistant/config_entries.py

```diff
@@ -1108,7 +1108,7 @@ async def _async_init_reconfigure(
                     "unique_id": self.unique_id,
                 }
                 | (context or {}),
-                data=self.data | (data or {}),
+                data=data or {},
```

Comment:
Do we even need to keep this code? It is never called as reconfigure is always triggered from fronted.